tests/test_unrolling.py

```diff
@@ -0,0 +1,67 @@
+"""Tests ``unrolling.py``."""
+
+import pytest
+
+from qibolab.pulses import Drag, Pulse, PulseSequence, PulseType, Rectangular
+from qibolab.unrolling import Bounds, batch
+
+
+def test_bounds_update():
+    p1 = Pulse(400, 40, 0.9, int(100e6), 0, Drag(5, 1), 3, PulseType.DRIVE)
+    p2 = Pulse(500, 40, 0.9, int(100e6), 0, Drag(5, 1), 2, PulseType.DRIVE)
+    p3 = Pulse(600, 40, 0.9, int(100e6), 0, Drag(5, 1), 1, PulseType.DRIVE)
+
+    p4 = Pulse(440, 1000, 0.9, int(20e6), 0, Rectangular(), 3, PulseType.READOUT)
+    p5 = Pulse(540, 1000, 0.9, int(20e6), 0, Rectangular(), 2, PulseType.READOUT)
+    p6 = Pulse(640, 1000, 0.9, int(20e6), 0, Rectangular(), 1, PulseType.READOUT)
+
+    ps = PulseSequence(p1, p2, p3, p4, p5, p6)
+    bounds = Bounds.update(ps)
+
+    assert bounds.waveforms == 40
+    assert bounds.readout == 3
+    assert bounds.instructions == 6
```

Comment:
Asserting specific numbers is a bit delicate, because the update mechanism may change.

Just go for reasonable lower bounds.

Reply:
But here I am just checking from an empty `Bounds`, we actually now the number of readouts on the sequence, if this gets changed I assume we should change the test and add other pulse sequences that reflect the different update ?

Reply:
The goal would be to be able to make a test that is also satisfied by a different implementation.

The goal of the tests in general is to check that everything behaves as expected. If you can avoid saying what you expect every time that you touch the code, there are fewer chances that your expectations will be driven by a faulty implementation :)

Reply:
So, for instructions and waveform duration it should be changed, but counting the number of readouts should be common to any implementation I guess.

Reply:
> counting the number of readouts should be common to any implementation I guess.

Yes and no: the number of readouts is a proxy for the amount of readout memory consumed. This of course will be greatly affected by the `AcquisitionType`. But I guess there is no way that any sensible test will survive that switch.